plugins/com.google.cloud.tools.eclipse.appengine.login.test/src/com/google/cloud/tools/eclipse/appengine/login/TransientOAuthDataStoreTest.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.google.cloud.tools.eclipse.appengine.login;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.tools.ide.login.OAuthData;
+
+import org.eclipse.e4.core.contexts.IEclipseContext;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TransientOAuthDataStoreTest {
+
+  @Mock IEclipseContext eclipseContext;
+
+  @Test
+  public void testLoadOAuthData_emptyStoreReturnsNonNullOAuthData() {
+    when(eclipseContext.get(anyString())).thenReturn(null);
+
+    OAuthData oAuthData = new TransientOAuthDataStore(eclipseContext).loadOAuthData();
+    Assert.assertNotNull(oAuthData);
+    Assert.assertEquals(null, oAuthData.getAccessToken());
+    Assert.assertEquals(null, oAuthData.getRefreshToken());
+    Assert.assertEquals(null, oAuthData.getStoredEmail());
+    Assert.assertEquals(0, oAuthData.getAccessTokenExpiryTime());
+  }
+
+  OAuthData singleStorageForIEclipseContext;
+
+  @Test
+  public void testSaveAndLoadOAuthData() {
```

Comment:
It's a test that `loadAuthData()` returns the `OAuthData` object set through `saveOAuthData()`. It took me some thing to make this work with Mockito. I couldn't find a better than the `Answer` setup.